specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json

```diff
@@ -2593,7 +2901,330 @@
+        }
+      ],
+      "description": "Radius client root certificate of VpnServerConfiguration."
+    },
+    "AadAuthenticationParameters": {
+      "properties": {
+        "aadTenant": {
+          "type": "string",
+          "description": "AAD Vpn authentication parameter AAD tenant."
+        },
+        "aadAudience": {
+          "type": "string",
+          "description": "AAD Vpn authentication parameter AAD audience."
+        },
+        "aadIssuer": {
+          "type": "string",
+          "description": "AAD Vpn authentication parameter AAD issuer."
+        }
+      },
+      "description": "AAD Vpn authentication type related parameters."
+    },
+    "VpnServerConfigurationProperties": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the VpnServerConfiguration that is unique within a resource group."
+        },
+        "vpnProtocols": {
+          "type": "array",
+          "items": {
+            "type": "string",
+            "description": "VPN protocol enabled for the VpnServerConfiguration.",
+            "enum": [
+              "IkeV2",
+              "OpenVPN"
+            ],
+            "x-ms-enum": {
+              "name": "VpnGatewayTunnelingProtocol",
+              "modelAsString": true
+            }
+          },
+          "description": "VPN protocols for the VpnServerConfiguration."
+        },
+        "vpnAuthenticationTypes": {
+          "type": "array",
+          "items": {
+            "type": "string",
+            "description": "VPN authentication types enabled for the VpnServerConfiguration.",
+            "enum": [
+              "Certificate",
+              "Radius",
+              "AAD"			  
+            ],
+            "x-ms-enum": {
+              "name": "VpnAuthenticationType",
+              "modelAsString": true
+            }
+          },
+          "description": "VPN authentication types for the VpnServerConfiguration."
+        },		
+        "vpnServerConfigVpnClientRootCertificates": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/VpnServerConfigVpnClientRootCertificate"
+          },
+          "description": "VPN client root certificate of VpnServerConfiguration."
+        },
+        "vpnServerConfigVpnClientRevokedCertificates": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/VpnServerConfigVpnClientRevokedCertificate"
+          },
+          "description": "VPN client revoked certificate of VpnServerConfiguration."
+        },
+        "vpnServerConfigRadiusServerRootCertificates": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/VpnServerConfigRadiusServerRootCertificate"
+          },
+          "description": "Radius Server root certificate of VpnServerConfiguration."
+        },
+        "vpnServerConfigRadiusClientRootCertificates": {
```

Comment:
A number of these certificate properties share a common schema, but each have a discrete definition. Will they change independently in the future? A common definition could be used.

Reply:
Certificates:- vpnServerConfigVpnClientRootCertificates and vpnServerConfigRadiusServerRootCertificates share common scheama and certificates:- vpnServerConfigVpnClientRevokedCertificates and vpnServerConfigRadiusClientRootCertificates share common schema. But, they represent different definitions e.g one is related to Certifiate vpn authentication and others are related to Radius vpn authentication so we want to keep them seperate.